test/run_test.py

```diff
@@ -50,146 +50,74 @@
-    "test_sparse_csr",
-    "test_quantization",
-    "test_pruning_op",
-    "test_spectral_ops",
-    "test_serialization",
-    "test_shape_ops",
-    "test_show_pickle",
-    "test_sort_and_select",
-    "test_tensor_creation_ops",
-    "test_testing",
-    "test_torch",
-    "test_type_info",
-    "test_unary_ufuncs",
-    "test_utils",
-    "test_view_ops",
-    "test_vmap",
-    "test_namedtuple_return_api",
-    "test_numpy_interop",
-    "test_jit_profiling",
-    "test_jit_legacy",
-    "test_jit_fuser_legacy",
-    "test_tensorboard",
-    "test_namedtensor",
-    "test_reductions",
-    "test_type_promotion",
-    "test_jit_disabled",
-    "test_function_schema",
-    "test_overrides",
-    "test_jit_fuser_te",
-    "test_tensorexpr",
-    "test_tensorexpr_pybind",
-    "test_openmp",
-    "test_profiler",
-    "distributed/test_launcher",
-    "distributed/nn/jit/test_instantiator",
-    "distributed/rpc/test_faulty_agent",
-    "distributed/rpc/test_tensorpipe_agent",
-    "distributed/rpc/cuda/test_tensorpipe_agent",
-    "test_determination",
-    "test_futures",
-    "test_fx",
-    "test_fx_experimental",
-    "test_functional_autograd_benchmark",
-    "test_package",
-    "test_license",
-    "distributed/pipeline/sync/skip/test_api",
-    "distributed/pipeline/sync/skip/test_gpipe",
-    "distributed/pipeline/sync/skip/test_inspect_skip_layout",
-    "distributed/pipeline/sync/skip/test_leak",
-    "distributed/pipeline/sync/skip/test_portal",
-    "distributed/pipeline/sync/skip/test_stash_pop",
-    "distributed/pipeline/sync/skip/test_tracker",
-    "distributed/pipeline/sync/skip/test_verify_skippables",
-    "distributed/pipeline/sync/test_balance",
-    "distributed/pipeline/sync/test_bugs",
-    "distributed/pipeline/sync/test_checkpoint",
-    "distributed/pipeline/sync/test_copy",
-    "distributed/pipeline/sync/test_deferred_batch_norm",
-    "distributed/pipeline/sync/test_dependency",
-    "distributed/pipeline/sync/test_inplace",
-    "distributed/pipeline/sync/test_microbatch",
-    "distributed/pipeline/sync/test_phony",
-    "distributed/pipeline/sync/test_pipe",
-    "distributed/pipeline/sync/test_pipeline",
-    "distributed/pipeline/sync/test_stream",
-    "distributed/pipeline/sync/test_transparency",
-    "distributed/pipeline/sync/test_worker",
-    "distributed/optim/test_zero_redundancy_optimizer",
-    "distributed/elastic/timer/api_test",
-    "distributed/elastic/timer/local_timer_example",
-    "distributed/elastic/timer/local_timer_test",
-    "distributed/elastic/events/lib_test",
-    "distributed/elastic/metrics/api_test",
-    "distributed/elastic/utils/logging_test",
-    "distributed/elastic/utils/util_test",
-    "distributed/elastic/utils/distributed_test",
-    "distributed/elastic/multiprocessing/api_test",
-    "distributed/_sharding_spec/test_sharding_spec",
-    "distributed/_sharded_tensor/test_sharded_tensor",
-]
+def discover_tests(
+        blocklisted_patterns: List[str] = [],
```

Comment:
can we add a top-level Path to this function?
```suggestion
def discover_tests(
        working_dir: Path,
        blocklisted_patterns: List[str] = [],
```

This would be useful to also replace the JIT_* and DISTRIBUTED_* test cases with globing by passing in `test/jit` and `test/distributed`, see #64253

Reply:
Sure, but let's do it as follow up PR. Also, to reduce the complexity we should rename some tests.

Reply:
Done